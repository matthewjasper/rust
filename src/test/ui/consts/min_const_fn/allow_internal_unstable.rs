// Check that allow_internal_unstable is not sufficient to allow features in
// min const fns.

#![feature(allow_internal_unstable)]

#[allow_internal_unstable(const_if_match)]
macro_rules! make_fn {
    () => {
        const fn get_number(x: bool) -> usize {
            if x { //~ ERROR loops and conditional expressions are not stable in const fn
                43
            } else {
                2765
            }
        }
    }
}

make_fn!();

fn main() {}
