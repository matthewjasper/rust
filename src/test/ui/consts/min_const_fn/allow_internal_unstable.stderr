error[E0723]: loops and conditional expressions are not stable in const fn
  --> $DIR/allow_internal_unstable.rs:10:13
   |
LL | /             if x {
LL | |                 43
LL | |             } else {
LL | |                 2765
LL | |             }
   | |_____________^
...
LL |   make_fn!();
   |   ----------- in this macro invocation
   |
   = note: for more information, see issue https://github.com/rust-lang/rust/issues/57563
   = help: add `#![feature(const_fn)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0723`.
